DataFormats/Portable/test/test_catch2_deepCopyOnHost.cc

```diff
@@ -0,0 +1,128 @@
+
+  // Portable Collection Views
+  SoAPositionView& positionCollectionView = positionCollection.view();
+  SoAPCAView& pcaCollectionView = pcaCollection.view();
+  // Portable Collection ConstViews
+  const SoAPositionConstView& positionCollectionConstView = positionCollection.const_view();
+  const SoAPCAConstView& pcaCollectionConstView = pcaCollection.const_view();
+
+  // fill up
+  for (size_t i = 0; i < elems; i++) {
+    positionCollectionView.x()[i] = static_cast<float>(i);
+    positionCollectionView.y()[i] = static_cast<float>(i) * 2.0f;
+    positionCollectionView.z()[i] = static_cast<float>(i) * 3.0f;
+  }
+  positionCollectionView.detectorType() = 1;
+
+  float time = 0.01;
+  for (size_t i = 0; i < elems; i++) {
+    pcaCollectionView.eigenvector_1()[i] = positionCollectionView.x()[i] / time;
+    pcaCollectionView.eigenvector_2()[i] = positionCollectionView.y()[i] / time;
+    pcaCollectionView.eigenvector_3()[i] = positionCollectionView.z()[i] / time;
+    pcaCollectionView[i].candidateDirection()(0) = positionCollectionView[i].x() / time;
+    pcaCollectionView[i].candidateDirection()(1) = positionCollectionView[i].y() / time;
+    pcaCollectionView[i].candidateDirection()(2) = positionCollectionView[i].z() / time;
+  }
+
+  SECTION("Deep copy the View") {
+    // addresses and size of the SoA columns
+    const auto posRecs = positionCollectionView.records();
+    const auto pcaRecs = pcaCollectionView.records();
+
+    // building the View with runtime check for the size
+    GenericSoAView genericView(posRecs.x(), posRecs.y(), posRecs.z(), pcaRecs.candidateDirection());
+
+    // Check for equality of memory addresses
+    REQUIRE(genericView.metadata().addressOf_xPos() == positionCollectionView.metadata().addressOf_x());
+    REQUIRE(genericView.metadata().addressOf_yPos() == positionCollectionView.metadata().addressOf_y());
+    REQUIRE(genericView.metadata().addressOf_zPos() == positionCollectionView.metadata().addressOf_z());
+    REQUIRE(genericView.metadata().addressOf_candidateDirection() ==
+            pcaCollectionView.metadata().addressOf_candidateDirection());
+
+    // PortableHostCollection that will host the aggregated columns
+    PortableHostCollection<GenericSoA> genericCollection(elems, cms::alpakatools::host());
+    genericCollection.deepCopy(genericView);
+
+    // Check for inequality of memory addresses
+    REQUIRE(genericCollection.view().metadata().addressOf_xPos() != positionCollectionView.metadata().addressOf_x());
+    REQUIRE(genericCollection.view().metadata().addressOf_yPos() != positionCollectionView.metadata().addressOf_y());
+    REQUIRE(genericCollection.view().metadata().addressOf_zPos() != positionCollectionView.metadata().addressOf_z());
+    REQUIRE(genericCollection.view().metadata().addressOf_candidateDirection() !=
+            pcaCollectionView.metadata().addressOf_candidateDirection());
+  }
+
+  SECTION("Aggregate the ConstView") {
```

Comment:
```suggestion
  SECTION("Deep copy the ConstView") {
```
for consistency with the previous section.